intelmq/bots/parsers/shadowserver/config.py

```diff
@@ -928,3 +936,303 @@ def validate_ip(value):
+    ],
+    'optional_fields': [
+        ('source.reverse_dns', 'hostname'),
+        ('event_description.text', 'tag'),
+        ('extra.', 'response', validate_to_none),
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+    ],
+    'constant_fields': {
+        'protocol.transport': 'udp',
+        'classification.type': 'vulnerable service',
+        'classification.identifier': 'opennetis',
+        'feed.code': 'shadowserver-opennetis',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/NTP-Version
+ntp_version = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+        ('source.port', 'port')
+    ],
+    'optional_fields': [
+        ('protocol.transport', 'protocol'),
+        ('source.reverse_dns', 'hostname'),
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('source.geolocation.region', 'region'),
+        ('source.geolocation.city', 'city'),
+        ('extra.', 'version', validate_to_none),
+        ('extra.', 'clk_wander', validate_to_none),
+        ('extra.', 'clock', validate_to_none),
+        ('extra.', 'error', validate_to_none),
+        ('extra.', 'frequency', validate_to_none),
+        ('extra.', 'jitter', validate_to_none),
+        ('extra.', 'leap', validate_to_none),
+        ('extra.', 'mintc', validate_to_none),
+        ('extra.', 'noise', validate_to_none),
+        ('extra.', 'offset', validate_to_none),
+        ('extra.', 'peer', validate_to_none),
+        ('extra.', 'phase', validate_to_none),
+        ('extra.', 'poll', validate_to_none),
+        ('extra.', 'precision', validate_to_none),
+        ('extra.', 'processor', validate_to_none),
+        ('extra.', 'refid', validate_to_none),
+        ('extra.', 'reftime', validate_to_none),
+        ('extra.', 'rootdelay', validate_to_none),
+        ('extra.', 'rootdispersion', validate_to_none),
+        ('extra.', 'stability', validate_to_none),
+        ('extra.', 'state', validate_to_none),
+        ('extra.', 'stratum', validate_to_none),
+        ('extra.', 'system', validate_to_none),
+        ('extra.', 'tai', validate_to_none),
+        ('extra.', 'tc', validate_to_none),
+    ],
+    'constant_fields': {
+        'classification.type': 'vulnerable service',
+        'classification.identifier': 'openntpversion',
+        'feed.code': 'shadowserver-openntpversion',
+        'protocol.application': 'ntp',
+    },
+}
+
+# https://www.shadowserver.org/wiki/pmwiki.php/Services/Sandbox-URL
+sandbox_url = {
+    'required_fields': [
+        ('time.source', 'timestamp', add_UTC_to_timestamp),
+        ('source.ip', 'ip'),
+    ],
+    'optional_fields': [
+        ('source.asn', 'asn'),
+        ('source.geolocation.cc', 'geo'),
+        ('malware.hash.md5', 'md5hash'),
+        ('event_description.url', 'url'),
```

Comment:
In this case probably Shadowserver get **IP** and **Host** from `url` field (tried to resolve some of domains, not all), so if we use **IP** as `source.ip`, **host** and **url** probably should be `source.fqdn` & `source.url` too?!